src/clib/pio_getput_int.c

```diff
@@ -307,6 +307,30 @@ int PIOc_put_att_tc(int ncid, int varid, const char *name, nc_type atttype,
 #ifdef _HDF5
     if (file->iotype == PIO_IOTYPE_HDF5)
     {
+        int num_attrs = file->hdf5_num_attrs;
+        if (num_attrs >= PIO_MAX_ATTRS)
+        {
+            fprintf(stderr, "ERROR: Num of attributes exceeds maximum (%d).\n", PIO_MAX_ATTRS);
```

Comment:
We might want to refrain from direct fprintfs and let pio_err handle the error message

Reply:
OK, ADIOS type has the same issue and we will fix it for both types.